src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_item.c

```diff
@@ -1,85 +1,1000 @@
+        false, // SLOT_BOW
+        false, // SLOT_ARROW_FIRE
+        false, // SLOT_ARROW_ICE
+        false, // SLOT_ARROW_LIGHT
+        true,  // SLOT_TRADE_DEED
+        false, // SLOT_BOMB
+        false, // SLOT_BOMBCHU
+        false, // SLOT_STICK
+        true,  // SLOT_NUT
+        false, // SLOT_MAGIC_BEANS
+        true,  // SLOT_TRADE_KEY_MAMA
+        false, // SLOT_POWDER_KEG
+        true,  // SLOT_PICTO_BOX
+        true,  // SLOT_LENS
+        false, // SLOT_HOOKSHOT
+        false, // SLOT_SWORD_GREAT_FAIRY
+        true,  // SLOT_TRADE_COUPLE
+        true,  // SLOT_BOTTLE_1
+        true,  // SLOT_BOTTLE_2
+        true,  // SLOT_BOTTLE_3
+        true,  // SLOT_BOTTLE_4
+        true,  // SLOT_BOTTLE_5
+        true,  // SLOT_BOTTLE_6
     },
+    // Human
     {
-        true,  true, true, true, true, true, true, true, true, true, true, true,
-        false, true, true, true, true, true, true, true, true, true, true, true,
+        true,  // SLOT_OCARINA
+        true,  // SLOT_BOW
+        true,  // SLOT_ARROW_FIRE
+        true,  // SLOT_ARROW_ICE
+        true,  // SLOT_ARROW_LIGHT
+        true,  // SLOT_TRADE_DEED
+        true,  // SLOT_BOMB
+        true,  // SLOT_BOMBCHU
+        true,  // SLOT_STICK
+        true,  // SLOT_NUT
+        true,  // SLOT_MAGIC_BEANS
+        true,  // SLOT_TRADE_KEY_MAMA
+        false, // SLOT_POWDER_KEG
+        true,  // SLOT_PICTO_BOX
+        true,  // SLOT_LENS
+        true,  // SLOT_HOOKSHOT
+        true,  // SLOT_SWORD_GREAT_FAIRY
+        true,  // SLOT_TRADE_COUPLE
+        true,  // SLOT_BOTTLE_1
+        true,  // SLOT_BOTTLE_2
+        true,  // SLOT_BOTTLE_3
+        true,  // SLOT_BOTTLE_4
+        true,  // SLOT_BOTTLE_5
+        true,  // SLOT_BOTTLE_6
     },
 };
 
-s16 D_8082B444[] = {
-    0x005F, 0x003E, 0x005F, 0x0080, 0x00A1, 0x00C2, 0x003E, 0x005F,
+s16 sAmmoRectLeft[] = {
+    95,  // SLOT_BOW
+    62,  // SLOT_BOMB
+    95,  // SLOT_BOMBCHU
+    128, // SLOT_STICK
+    161, // SLOT_NUT
+    194, // SLOT_MAGIC_BEANS
+    62,  // SLOT_POWDER_KEG
+    95,  // SLOT_PICTO_BOX
 };
 
-s16 D_8082B454[] = {
-    0x0055, 0x0075, 0x0075, 0x0075, 0x0075, 0x0075, 0x0096, 0x0096, 0x00FF, 0x0064,
-    0x00FF, 0x0000, 0x0000, 0x0064, 0x00FF, 0x0000, 0x0000, 0x00FF, 0x0064, 0x0000,
+s16 sAmmoRectHeight[] = {
+    85,  // SLOT_BOW
+    117, // SLOT_BOMB
+    117, // SLOT_BOMBCHU
+    117, // SLOT_STICK
+    117, // SLOT_NUT
+    117, // SLOT_MAGIC_BEANS
+    150, // SLOT_POWDER_KEG
+    150, // SLOT_PICTO_BOX
 };
 
-u8 D_8082B47C[] = {
-    0x35, 0x33, 0x34, 0x32, 0xFF,
-};
+void KaleidoScope_DrawAmmoCount(PauseContext* pauseCtx, GraphicsContext* gfxCtx, s16 item, u16 ammoIndex) {
+    s16 ammoUpperDigit;
+    s16 ammo;
 
-s16 D_8082B484[] = {
-    0x0294,
-    0x0384,
-    0x0474,
-    0x0000,
-};
+    OPEN_DISPS(gfxCtx);
+
+    if (item == ITEM_PICTO_BOX) {
+        if (!(((void)0, gSaveContext.save.inventory.questItems) & (gBitFlags[QUEST_PICTOGRAPH]))) {
```

Comment:
Isn't this the `CHECK_QUEST_ITEM` macro?
```suggestion
        if (!CHECK_QUEST_ITEM(QUEST_PICTOGRAPH)) {
```